src/mame/drivers/blit.cpp

```diff
@@ -0,0 +1,312 @@
+// license:BSD-3-Clause
+// copyright-holders:Sergey Svishchev
+/***************************************************************************
+
+    AT&T Blit -- monochrome, raster graphics (800x1024), serial terminal.
+
+    https://code.9front.org/hg/plan9front/file/f4fa0b9d0397/sys/src/games/blit/mmap
+        memory map
+
+    https://pbs.twimg.com/media/C_uxVGgUwAAImk6.jpg:orig
+        board photo
+
+    Onboard hardware (common to all revisions) is
+    - 24K ROM
+    - 256K RAM (includes frame buffer)
+    - 2 serial ports, each driven by 6850 ACIA
+
+    To do:
+    - keyboard
+    - mouse
+    - make 'mux' window system work -- "68ld: load protocol failed"
+    - sound?
+    - what do dip switches do?
+
+****************************************************************************/
+
+#include "emu.h"
+
+#include "bus/rs232/rs232.h"
+#include "cpu/m68000/m68000.h"
+#include "machine/6850acia.h"
+#include "machine/clock.h"
+#include "machine/ram.h"
+
+#include "emupal.h"
+#include "screen.h"
+#include "speaker.h"
+
+
+#define M68K_TAG "maincpu"
+#define ACIA0_TAG "acia0"
+#define ACIA1_TAG "acia1"
+#define RS232_H_TAG "host"
+#define RS232_K_TAG "keyboard"
+
+
+//#define LOG_GENERAL (1U <<  0) //defined in logmacro.h already
+#define LOG_PIA       (1U <<  1)
+#define LOG_DEBUG     (1U <<  2)
+
+//#define VERBOSE (LOG_DEBUG)
+//#define LOG_OUTPUT_FUNC printf
+#include "logmacro.h"
+
+#define LOGPIA(...) LOGMASKED(LOG_PIA, __VA_ARGS__)
+#define LOGDBG(...) LOGMASKED(LOG_DEBUG, __VA_ARGS__)
+
+
+class blit_state : public driver_device
+{
+public:
+	blit_state(const machine_config &mconfig, device_type type, const char *tag)
+		: driver_device(mconfig, type, tag)
+		, m_maincpu(*this, M68K_TAG)
+		, m_ram(*this, RAM_TAG)
+		, m_acia0(*this, ACIA0_TAG)
+		, m_acia1(*this, ACIA1_TAG)
+		, m_screen(*this, "screen")
+		, m_misccr(*this, "misccr")
+		, m_p_ram(*this, "p_ram")
+		, m_sysrom(*this, M68K_TAG)
+	{ }
+
+	static constexpr feature_type imperfect_features() { return feature::KEYBOARD; }
+
+	DECLARE_WRITE_LINE_MEMBER(system_clock_write);
+
+	uint16_t bootvect_r(offs_t offset);
+	void bootvect_w(offs_t offset, uint16_t data, uint16_t mem_mask = ~0);
+	void start_write(offs_t offset, uint16_t data, uint16_t mem_mask = ~0);
+	void vblank_ack(offs_t offset, uint16_t data, uint16_t mem_mask = ~0);
+
+	uint32_t screen_update(screen_device &screen, bitmap_ind16 &bitmap, const rectangle &cliprect);
+
+	void blit(machine_config &config);
+	void blit_mem(address_map &map);
+
+private:
+	virtual void machine_start() override;
+	virtual void machine_reset() override;
```

Comment:
Overridden protected members should be protected.  The idea being that if someone subclasses this for a related system, it’s easy to call through to the base members.